samples/XAMLCatalog/XAMLCatalog/UIKitControls/UIButtonWithControlsViewController.m

```diff
@@ -64,9 +64,35 @@ - (UIControlState)getControlState:(NSString*)stateString {
     return state;
 }
 
+- (UIEdgeInsets)getUIEdgeInsetsFromString:(NSString*)edgeInsetsString {
+    UIEdgeInsets ret = UIEdgeInsetsZero;
+
+    if ([edgeInsetsString length] > 0) {
+        // Strip off curly braces and use the delimiter to separate components
+        NSString* strWithBraces = [edgeInsetsString substringWithRange : NSMakeRange(1, edgeInsetsString.length - 1)];
+        NSArray* components = [strWithBraces componentsSeparatedByString : @","];
+        for (int componentIndex = 0; componentIndex < 4; componentIndex++) {
+            CGFloat value = ((NSString*)[components objectAtIndex : componentIndex]).floatValue;
+            if (componentIndex == 0) {
+                ret.top = value;
+            } else if (componentIndex == 1) {
+                ret.left = value;
+            } else if (componentIndex == 2) {
+                ret.bottom = value;
+            } else if (componentIndex == 3) {
+                ret.right = value;
+            }
```

Comment:
Since this is an input field, we should draw attention to the invalid value by turning it red or some standard UI validation approach (! etc)

---
In reply to: [104816209](https://github.com/Microsoft/WinObjC/pull/2156#discussion_r104816209) [](ancestors = 104816209)